app/utils/geostore.py

```diff
@@ -51,3 +73,54 @@ async def get_geostore(geostore_id: UUID, geostore_origin: str) -> GeostoreCommo
         raise HTTPException(status_code=500, detail=str(e))
     except BadResponseError as e:
         raise HTTPException(status_code=400, detail=str(e))
+
+
+async def get_geostore_from_any_origin(
+    geostore_id: UUID, geostore_origin: GeostoreOrigin
+) -> GeostoreCommon:
+    """Looks for geometry in all geostores, beginning with client's choice."""
+
+    geostore_constructor = {
+        GeostoreOrigin.gfw: _get_gfw_geostore,
+        GeostoreOrigin.rw: rw_api.get_geostore,
+    }
+
+    geo_func = geostore_constructor.pop(geostore_origin)
+
+    # If we get a geostore from any origin return it,
+    # if we get all 404s return a 404,
+    # if we get a mixture of errors return a 500 with explanation
```

Comment:
Nice! Does this need to make it into the rdocs, @dmannarino?

Reply:
Mmm, good point. Will add it to the route documentation.